src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -565,6 +571,28 @@ def resolve_imports_and_packages(
             max_batch_size,
             source_code_display=source_code_display,
         )
+
+        if pickled_by_reference:
+            try:
+                source_code = code_generation.generate_source_code(func, False)
+            except Exception as e:
+                raise RuntimeError(
+                    f"Could not register udf, source code extraction failed for function pickled by reference: {e}"
+                )
+            with tempfile.TemporaryDirectory() as tempdir:
+                abs_path = tempdir + f"/{func.__module__.replace('.', '/')}.py"
```

Comment:
If we create two udf, there should be two separate files written under different temp directory locally, and the files will be uploaded to separate stages. I think there shouldn't be a conflict?

Reply:
ah ok I didn't realize that the temp directories are different. But should we still add function_name to its filename?

Reply:
We cannot do that since we need the file to be named `{func.__module__}.py`, which fixes the `ModuleNotFound` error.

Reply:
I see. Could you test what if two files with the same name are included in import clause in SQL of creating UDF. I remember it will fail (even with different path).

Reply:
You are right, it would fail with `SQL compilation error: Duplicate file name notz.parquet in IMPORTS clause.`
When would this cause happen tho? If the user specifies a import also named `{func.__module__}.py`?

Reply:
yea, I just feel `{func.__module__}.py` is kinda fragile